### DISCUSSION TITLE
Show more pages around current page in stepped pagination

### DISCUSSION THREAD

**Discussion by Azarchaniel** [material-table-core/core#477] [Ideas]
I would like to have option to set a number of pages 'around' current page.
Example:
Currently, when I am on page 6, I see something like this < 5 6 7 >.
I would like to have an option, like {numberOfPagesAroundCurrent: 2}, and the result will be < 4 5 6 7 8 >.
{numberOfPagesAroundCurrent: 3} will lead to < 3 4 5 6 7 8 9 >.
Of course, when we are getting to the last page, it should be 'cropped'. The last page is 10, the current is 8 and numberOfPagesAroundCurrent is 3. The result therefore would be < 5 6 7 8 9 10 >. Same for the first page.
Optional parameter, default would be 1 like now.

**Comment by Domino987** [comment]
This would be a nice addition. The fastest would be to just override the pagination for component for you. You could also create a PR to add additional numbers.

**Reply by Azarchaniel** [reply]
https://github.com/mbrn/material-table/pull/3112

**Reply by Domino987** [reply]
Would you mind porting this to this repo ?

**Reply by Azarchaniel** [reply]
shit, I knew I will make some mistake! 
I'll look at it.

**Reply by Azarchaniel** [reply]
I rewrote it to this repo, but I'm unable to do pull request.

**Reply by Domino987** [reply]
Do you know why? Because I know that it works

**Reply by Azarchaniel** [reply]
remote: Permission to material-table-core/core.git denied to Azarchaniel.
fatal: unable to access 'https://github.com/material-table-core/core.git/': The requested URL returned error: 403

**Reply by Azarchaniel** [reply]
I have a branch "feature/numberOfPagesAround".

**Reply by Domino987** [reply]
Ok that's a commit not PR aka just a code push. You need to fork the repo, add the changes to a branch in your fork and from that fork create a PR to the base repo. Afterwards it should be listed in the base repo pull request tab for me to review.

**Reply by Domino987** [reply]
Can you show me the link to it? Or this git ls-remote --get-url

**Reply by Domino987** [reply]
You might just have cloned it, aka copied the code to your machine.

**Reply by Azarchaniel** [reply]
Hold on, I'm writing description to PR.

**Reply by Azarchaniel** [reply]
Try this? 
https://github.com/material-table-core/core/pull/501

**Reply by Azarchaniel** [reply]
Can you add this option to the docs, please?

**Reply by Domino987** [reply]
Yes this will be the last step
